pyccel/codegen/printing/fcode.py

```diff
@@ -1408,32 +1402,65 @@ def _print_FunctionDef(self, expr):
             sig = 'elemental {}'.format(sig)
 
         arg_code  = ', '.join(self._print(i) for i in chain( expr.arguments, out_args ))
+
+        imports = ''.join(self._print(i) for i in expr.imports)
+        prelude = ''.join(self._print(i) for i in args_decs.values())
+        parts = ["{}({}) {}\n".format(sig, arg_code, func_end,),
+                imports,
+                "implicit none\n",
```

Comment:
This line can be removed: since there are no local variables in a function definition, there is no danger of an implicit type declaration (Fortran-77 style)